synapse/handlers/room.py

```diff
@@ -681,6 +681,15 @@ async def create_room(
             creator_id=user_id, is_public=is_public, room_version=room_version,
         )
 
+        # Check whether this visibility value is blocked by a third party module
+        allowed_by_third_party_rules = await (
+            self.third_party_event_rules.check_visibility_can_be_modified(
```

Comment:
think this needs an update now?

Reply:
Not sure what you mean? It should be based on the desired room visibility in the public room list rather than aliases as it was before.

Reply:
Oh, you probably meant the call to `check_room_can_be_added_to_public_rooms_directory`. Yes, that needed updating, thanks.